src/devices/OneWire/OneWireController.Linux.cs

```diff
@@ -0,0 +1,55 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Threading.Tasks;
+
+namespace Iot.Device.OneWire
+{
+    public partial class OneWireController
+    {
+        internal const string SysfsBusDevicesPath = "/sys/bus/w1/devices";
+        internal const string SysfsDevicesPath = "/sys/devices";
+
+        private static IEnumerable<OneWireBus> EnumerateBusesInternal()
+        {
+            foreach (var entry in Directory.EnumerateDirectories(SysfsBusDevicesPath, "w1_bus_master*"))
+            {
+                yield return new OneWireBus(Path.GetFileName(entry));
+            }
+        }
+
+        internal static IEnumerable<OneWireDevice> EnumerateDevices(OneWireBus bus, OneWireBus.DeviceFamily family)
```

Comment:
I think this should be moved as a static method of OneWireDevice, enumerating buses to OneWireBus and this class should be removed